### PR TITLE
Add `dolt reset HEAD~` 

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
This pr addresses the first part of this issue #1492

### CODE REVIEW

File: integration-tests/bats/status.bats [hunk 1]
```diff
@@ -204,4 +204,95 @@ SQL
 @test "status: dolt reset --hard with more than one additional arg throws an error " {
     run dolt reset --hard HEAD HEAD2
     [ "$status" -eq 1 ]
+}
+
+@test "status: dolt reset hard with ~ works" {
```
Comment (timsehn) [inline integration-tests/bats/status.bats:209]: I think you should test other refspecs. Like a branch name, a commit hash, in addition to just ~

**Review by timsehn — COMMENTED** [review]
Bats comment

File: integration-tests/bats/status.bats (hunk 1 above)
Comment (VinaiRachakonda) [reply integration-tests/bats/status.bats:209]: Good point. Knew I was forgetting something

File: go/libraries/doltcore/env/actions/reset.go [hunk 2]
```diff
@@ -255,3 +287,19 @@ func resetStaged(ctx context.Context, ddb *doltdb.DoltDB, rsw env.RepoStateWrite
...
+func ValidateIsRef(ctx context.Context, cSpecStr string, ddb *doltdb.DoltDB, rsr env.RepoStateReader) bool {
+	cs, err := doltdb.NewCommitSpec(cSpecStr)
+	if err != nil {
+		return false
+	}
+
+	_, err = ddb.Resolve(ctx, cs, rsr.CWBHeadRef())
+	if err != nil {
+		return false
+
```
Comment (zachmu) [inline go/libraries/doltcore/env/actions/reset.go:301]: Useless blank line

File: integration-tests/bats/status.bats [hunk 3]
```diff
@@ -204,4 +204,95 @@ SQL
...
+    [[ "$output" =~ "NULL" ]] || false
+
+    # Since this is a hard reset double check the status
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "On branch master" ]] || false
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+}
+
+@test "status: dolt reset soft with ~ works" {
```
Comment (zachmu) [inline integration-tests/bats/status.bats:252]: You have logic in the command about updating staged roots, but you aren't testing any of that logic here. Add some plz

**Review by zachmu — APPROVED** [review]
LG, just a little test expansion

**Review by timsehn — APPROVED** [review]
BATS looks great now. Well done.

**Pull request merged**